plugins/modules/nmcli.py

```diff
@@ -2130,8 +2137,12 @@ def _compare_conn_params(self, conn_info, options):
 
             if isinstance(current_value, list) and isinstance(value, list):
                 # compare values between two lists
-                if sorted(current_value) != sorted(value):
-                    changed = True
+                if key in ('ipv4.addresses', 'ipv6.addresses'):
+                    # The order of IP addresses matters because the first one
+                    # is the default source address for outbound connections.
+                    changed = current_value != value
+                else:
+                    changed = sorted(current_value) != sorted(value)
```

Comment:
This part seems to fail the unit tests.

Not sure why...I would suggest to move this to another PR and make the origin changes for this PR commited?

Reply:
Actually this test shows a bug in the above code: the version before set `changed = True` when there was a change. The version now **always sets `changed` to `True` or `False`**. So if `changed` was `True`, it might now have been set to `False`.

Reply:
(Basically this can happen as soon as the loop in 2102 has more than one iteration.)

Reply:
Dammit that'll teach me to try to sneak a new fix in at the last minute. My bad.
Yes, I have moved this change to another branch and I will both fix the logic and make sure the unit tests pass before submitting it. Sorry for the churn.